### PR TITLE
Made the static signature of 'size' more precise

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala [hunk 1]
```diff
@@ -349,12 +349,21 @@ class WdlStandardLibraryFunctionsType extends WdlFunctions[WdlType] {
   def write_objects(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlFileType)
   def write_json(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlFileType)
   def glob(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlArrayType(WdlFileType))
-  def size(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlFloatType)
+  def isGoodFirstSizeParam(wdlType: WdlType): Boolean = wdlType match {
```
Comment (geoffjentry) [inline src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala]: nitpick, but while i kinda understand why all of the engine functions are crammed together like this, cramming multiline helper functions in there seems less aesthetically pleasing to me.

Also, is this intended to be overridden or something? If not maybe make it private?

File: src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala [hunk 2]
```diff
@@ -349,12 +349,21 @@ class WdlStandardLibraryFunctionsType extends WdlFunctions[WdlType] {
...
+    case Success(f) :: Nil if isGoodFirstSizeParam(f) => Success(WdlFloatType)
+    case Success(f) :: Success(WdlStringType) :: Nil if isGoodFirstSizeParam(f) => Success(WdlFloatType)
+    case other => Failure(new Exception(s"Unexpected arguments to function `size`. Expected 'size(file: File, [unit: String])' but got 'size(${other.map(_.map(_.toWdlString)).mkString(", ")})'"))
+  }
   def length(params: Seq[Try[WdlType]]): Try[WdlType] = params.toList match {
     case Success(WdlArrayType(_)) :: Nil => Success(WdlIntegerType)
     case _ =>
       val badArgs = params.mkString(", ")
-      Failure(new Exception(s"Unexpected arguments to function `length`.  `length` takes a parameter of type Array but got: $badArgs"))
+      Failure(new Exception(s"Unexpected arguments to function `length`. `length` takes a parameter of type Array but got: $badArgs"))
```
Comment (mcovarr) [inline src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala]: 👾

File: src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala (hunk 1 above)
Comment (mcovarr) [inline src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala]: or put it in `size`

File: src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala [hunk 3]
```diff
@@ -349,12 +349,21 @@ class WdlStandardLibraryFunctionsType extends WdlFunctions[WdlType] {
...
-  def size(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlFloatType)
+  def isGoodFirstSizeParam(wdlType: WdlType): Boolean = wdlType match {
+    case f if WdlFileType.isCoerceableFrom(f) => true
+    case WdlOptionalType(o) => isGoodFirstSizeParam(o)
+    case _ => false
+  }
+  def size(params: Seq[Try[WdlType]]): Try[WdlType] = params.toList match {
+    case Success(f) :: Nil if isGoodFirstSizeParam(f) => Success(WdlFloatType)
+    case Success(f) :: Success(WdlStringType) :: Nil if isGoodFirstSizeParam(f) => Success(WdlFloatType)
+    case other => Failure(new Exception(s"Unexpected arguments to function `size`. Expected 'size(file: File, [unit: String])' but got 'size(${other.map(_.map(_.toWdlString)).mkString(", ")})'"))
```
Comment (mcovarr) [inline src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala]: `, [unit: String]` => `[, unit: String]`

**Comment by mcovarr** [comment]
:+1:

**Comment by danbills** [comment]
👍

**Pull request merged**